### PR TITLE
feat: implement ui-icon component

### PR DESCRIPTION
**Pull request by beeman**

### CODE REVIEW

**Comment by nartc** [comment]
LGTM! Let's try to merge this before my PR :D

**Review by nartc — APPROVED** [review]

**Comment by beeman** [comment]
I'm going to give it another look tomorrow. There's are quite some things that needed to be added (like `flex items-center`) that I'm not really happy with.

Also, we'd still need font-awesome for doing the input addons, I'm going to check if I can convert those addons to using the `ui-icon` library, then get rid of font-awesome all together.

Also, maybe we should implement [ngneat/svg-icon](https://github.com/ngneat/svg-icon) as it comes with sizing out of the box.

image

**Comment by nartc** [comment]
Interesting. Yeah, let's give ngneat/svg-icon a go

**Comment by ruslanguns** [comment]
This project seems interesting! Do you guys have a roadmap or something to contribute with you?

**Comment by beeman** [comment]
Hey @ruslanguns, thanks for your message! We don't currently have a roadmap, though we do have some ideas on what we like to do.

If you'd like to add something let's chat on Discord to talk about the details :)

**Comment by ruslanguns** [comment]
Sure!

File: src/app/features/workflows/components/workflow-group/workflow-group.component.ts [hunk 1]
```diff
@@ -10,15 +10,15 @@ import { WorkflowGroupStore } from './workflow-group.store'
         <h2
           class="py-2 px-2 dark:bg-gray-700 dark:text-gray-300 bg-gray-400 text-gray-900 dark:text-gray-100 flex justify-between"
         >
-          <div>
+          <div class="flex items-center">
             <button class="text-gray-700 dark:text-gray-300 mr-2" (click)="toggleCollapse()">
-              <i class="fa fa-fw {{ !vm.isCollapsed ? 'fa-minus-circle' : 'fa-plus-circle' }}"></i>
+              <ui-icon size="md" [icon]="vm.isCollapsed ? 'plusCircle' : 'minusCircle'"></ui-icon>
```
Comment (nartc) [inline src/app/features/workflows/components/workflow-group/workflow-group.component.ts:15]: nitpick (non-blocking): I am super tempted to return a `collapseIcon` from the `vm`.

Comment (nartc) [reply src/app/features/workflows/components/workflow-group/workflow-group.component.ts:15]: nitpick: isn't `ui-icon` default `size` `md`?

Comment (beeman) [reply src/app/features/workflows/components/workflow-group/workflow-group.component.ts]: Thanks! Great points, just implemented them.

**Pull request merged**